File: api/models/card.go
package models

import (
	"github.com/BackendApiCardExam/api/config"
	"gorm.io/gorm"
)

var db *gorm.DB

type Card struct {
	gorm.Model
	Name             string           `json:"Name"`
	University       string           `json:"University"`
	Direction        string           `json:"Direction"`
	Course           string           `json:"Course"`
	Description      string           `json:"Description"`
	CreatorID        int              `json:"CreatorId"`
	Rating           int              `json:"Rating"`
	ViewersID        int              `json:"ViewersId"`
	Comments         []Comment        `json:"Comments"`
	DateExam         string           `json:"DateExam"`
	IsVisible        bool             `json:"IsVisible"`
	TypeCard         int              `json:"TypeCard"`
	QuestionsAnswers []QuestionAnswer `json:"QuestionsAnswers"`
}

type Comment struct {
	gorm.Model
	IDUser int    `json:"IdUser"`
	Text   string `json:"Text"`
	CardID int
}

type QuestionAnswer struct {
	gorm.Model
	Question string `json:"Question"`
	Answer   string `json:"Answer"`
	CardID   int
}

func init() {
	config.Connect()
	db = config.GetDB()
	db.AutoMigrate(&Card{}, &Comment{}, &QuestionAnswer{})
}

func (c *Card) CreateCardToDb() *Card {
	db.Create(&c)
	return c
}

func GetCardByIdFromDb(Id int64) (*Card, *gorm.DB) {
	var getCard Card
	db.Preload("QuestionsAnswers").Preload("Comments").Find(&getCard, "ID = ?", Id)
	return &getCard, db
}

File: lib/database/card.dart
import 'package:cardexam/database/questionanswer.dart';
import 'package:hive_flutter/hive_flutter.dart';

part 'card.g.dart';

@HiveType(typeId: 2)
class Card {
  Card({
    required this.id,
    required this.name,
    required this.universe,
    required this.direction,
    required this.course,
    required this.description,
    required this.creatorId,
    required this.rating,
    required this.dateExam,
    required this.isVisible,
    required this.questionAnswers,
  });
  // Id карточки
  @HiveField(0)
  int id;
  // Название карточки
  @HiveField(1)
  final String name;
  // Название университета
  @HiveField(2)
  final String universe;
  // Направление
  @HiveField(3)
  final String direction;
  // Курс
  @HiveField(4)
  final int course;
  // Описание
  @HiveField(5)
  final String description;
  // Создатель карточки
  @HiveField(6)
  final int creatorId;
  // Рейтинг карточки
  @HiveField(7)
  final double rating;
  // Дата экзамена
  @HiveField(8)
  final DateTime dateExam;
  // Видна ли карточка в общем доступе
  @HiveField(9)
  final bool isVisible;
  // Список вопрос-ответов
  @HiveField(10)
  final List<QuestionAnswer> questionAnswers;
}
